obal/data/modules/rpmspec_changelog_check.py

```diff
@@ -7,23 +7,49 @@
 import glob
 
 ANSIBLE_METADATA = {
-    'metadata_version': '1.1',
+    'metadata_version': '1.2',
     'status': ['preview'],
     'supported_by': 'community'
 }
 
 from ansible.module_utils.basic import AnsibleModule  # pylint: disable=C0413
 
+def get_changelog_evr(specfile):
+    evr = subprocess.check_output([
+        'rpm',
+        '--query',
+        '--changelog',
+        '--specfile',
+        specfile
+    ])
+    return evr.split("\n")[0].split(" ")[-1]
+
+def get_specfile_evr(specfile):
```

Comment:
```python
>>> get_specfile_evr('/home/remote/egolov/Devel/theforeman/foreman-packaging/packages/foreman/foreman/foreman.spec')
'1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop1.20.0-0.10.develop'
```

this looks wrong :)

Reply:
mhh, pure python:

```
>>> import rpm
>>> rpm.delMacro('dist')
>>> s = rpm.spec('/home/remote/egolov/Devel/theforeman/foreman-packaging/packages/foreman/foreman-installer/foreman-installer.spec')
>>> s.sourceHeader['evr']
'1:1.20.0-0.1.develop'
>>> s = rpm.spec('/home/remote/egolov/Devel/theforeman/foreman-packaging/packages/foreman/foreman/foreman.spec')
>>> s.sourceHeader['evr']
'1.20.0-0.10.develop'
```

be aware, that on python3 this returns `bytes`, not `str`, so it's safest to use 

```
>>> import rpm
>>> rpm.delMacro('dist')
>>> s = rpm.spec('/home/remote/egolov/Devel/theforeman/foreman-packaging/packages/foreman/foreman-installer/foreman-installer.spec')
>>> s.sourceHeader['evr']
b'1:1.20.0-0.1.develop'
>>> s.sourceHeader['evr'].decode('ascii')
'1:1.20.0-0.1.develop'
```

(that should work on python2 too)

Reply:
That's a fancy use of library code